### PR TITLE
INTENG 5176 allow client control over events

### PR DESCRIPTION
**Pull request by echo-branch**
Add a public API to customize event sharing.  This feature can be tested by using the sample app with a break point on the callback.

### CODE REVIEW

File: AdobeBranchExtension/Classes/AdobeBranchExtensionClass.m [hunk 1]
```diff
@@ -67,17 +68,32 @@ + (BOOL)application:(UIApplication *)application openURL:(NSURL *)url options:(N
...
     if (!self) return self;
 
     BNCLogSetDisplayLevel(BNCLogLevelError);
 
     NSError *error = nil;
-    if ([self.api registerWildcardListener:AdobeBranchExtensionListener.class error:&error])
+    if ([self.api registerWildcardListener:AdobeBranchExtensionListener.class error:&error]) {
         BNCLogDebug(@"BranchExtensionRuleListener was registered.");
-    else
+    } else {
```
Comment (csalmi-branch) [inline AdobeBranchExtension/Classes/AdobeBranchExtensionClass.m]: Seems like an odd thing to need to add. Do we have a style guide that we could enforce?

File: AdobeBranchExtension/Classes/AdobeBranchExtensionClass.m [hunk 2]
```diff
@@ -126,7 +142,7 @@ - (void)handleEvent:(ACPExtensionEvent*)event {
     return dictionary;
 }
 
-+ (BranchEvent *)branchEventFromAdbobeEventName:(NSString *)eventName
++ (BranchEvent *)branchEventFromAdobeEventName:(NSString *)eventName
```
Comment (csalmi-branch) [inline AdobeBranchExtension/Classes/AdobeBranchExtensionClass.m]: wow, that was a neat typo! Good catch.

**Comment by aaaronlopez** [comment]
👍 nice, thanks for owning this feature request so quickly!

**Comment by aaaronlopez** [comment]
cc @jgeraghty @Jgilhuly

**Pull request merged**